modules/layers/src/solid-polygon-layer/polygon-tesselator.js

```diff
@@ -110,11 +111,11 @@ export default class PolygonTesselator extends Tesselator {
     let i = indexStart;
 
     // 1. get triangulated indices for the internal areas
-    const indices = Polygon.getSurfaceIndices(polygon);
+    const indices = Polygon.getSurfaceIndices(polygon, this.positionSize);
 
     // make sure the buffer is large enough
     if (currentLength < i + indices.length) {
-      currentLength *= 2;
+      currentLength = (i + indices.length) * 2;
```

Comment:
Could you explain why do we need the `i +` here?
target with a larger indexStart will need large buffer size?

Reply:
The original strategy was to double the current buffer, but it did not account for the case where double was still not large enough. This seems a safe way to expand the current buffer (to twice the needed size) without using a while loop.